src/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.c

```diff
@@ -175,6 +176,20 @@ PAL_X509ContentType AppleCryptoNative_X509GetContentType(uint8_t* pbData, int32_
         }
     }
 
+    dataFormat = kSecFormatPEMSequence;
+    actualFormat = dataFormat;
+    itemType = kSecItemTypeAggregate;
+    actualType = itemType;
+
+    if (osStatus == noErr)
```

Comment:
Isn't there a missing call to SecItemImport here?

Reply:
(Assuming that things work because reading as kSecFormatX509Cert/kSecItemTypeCertificate returned noErr and changed to PEMSequence/Aggregate, this if should be merged with the previous one... but as it is, dataFormat and actualFormat (and itemType/actualType) will always match :smile:)

Reply:
Too much Cmd+Z 😔. One of these days I will make a PR that will show evidence that I am capable of reading.

Reply:
>this if should be merged with the previous one

There are a few of `if { if { ... }}` blocks up above. Should all of them be addressed?

Reply:
Sorry, by "merged with the previous" I meant a second if inside the previous `if (osStatus == noErr)`:

```C
    dataFormat = kSecFormatX509Cert;
    actualFormat = dataFormat;
    itemType = kSecItemTypeCertificate;
    actualType = itemType;

    osStatus = SecItemImport(cfData, NULL, &actualFormat, &actualType, 0, NULL, NULL, NULL);

    if (osStatus == noErr)
    {
        if ((actualType == itemType && actualFormat == dataFormat) ||
            (actualType == kSecItemTypeAggregate && actualFormat == kSecFormatPEMSequence))
        {
            CFRelease(cfData);
            return PAL_Certificate;
        }
    }
```